tests/src/main/java/net/neoforged/neoforge/debug/capabilities/handlers/legacy/LegacyFluidUtilTest.java

```diff
@@ -0,0 +1,129 @@
+        helper.assertTrue(simulateResult.isSuccess(), "Failed to transfer.");
+        checkItemStack(helper, simulateResult.getResult(), Items.BUCKET, 1);
+        // Tank and stack shouldn't be modified
+        checkItemStack(helper, sourceStack, Items.WATER_BUCKET, 2);
+        checkFluidStack(helper, targetTank.getFluid(), Fluids.EMPTY, 0);
+
+        // Execute should modify
+        var executeResult = FluidUtil.tryEmptyContainer(sourceStack, targetTank, 1000, null, true);
+        helper.assertTrue(executeResult.isSuccess(), "Failed to transfer.");
+        checkItemStack(helper, executeResult.getResult(), Items.BUCKET, 1);
+        checkFluidStack(helper, targetTank.getFluid(), Fluids.WATER, 1000);
+        checkItemStack(helper, sourceStack, Items.WATER_BUCKET, 2); // Apparently the stack is not supposed to be modified
+        helper.succeed();
+    }
+
+    /**
+     * Ensures that tryFillContainer doesn't change the target fluid handler when simulating.
+     * Ant that the result of the simulated transfver is valid.
+     * Regression test for the root cause of <a href="https://github.com/MinecraftForge/MinecraftForge/issues/6796">issue #6796</a>.
+     */
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Legacy tryFillContainer")
+    private static void test_tryFillContainer(ExtendedGameTestHelper helper) {
+        var targetStack = new ItemStack(Items.BUCKET, 2);
+        var sourceTank = new FluidTank(10000);
+        sourceTank.setFluid(new FluidStack(Fluids.WATER, 5000));
+
+        // Simulate is not supposed to modify anything
+        var simulateResult = FluidUtil.tryFillContainer(targetStack, sourceTank, 1000, null, false);
+        helper.assertTrue(simulateResult.isSuccess(), "Failed to transfer.");
+        checkItemStack(helper, simulateResult.getResult(), Items.WATER_BUCKET, 1);
+        // Tank and stack shouldn't be modified
+        checkItemStack(helper, targetStack, Items.BUCKET, 2);
+        checkFluidStack(helper, sourceTank.getFluid(), Fluids.WATER, 5000);
+
+        // Execute should modify
+        var executeResult = FluidUtil.tryFillContainer(targetStack, sourceTank, 1000, null, true);
+        helper.assertTrue(executeResult.isSuccess(), "Failed to transfer.");
+        checkItemStack(helper, executeResult.getResult(), Items.WATER_BUCKET, 1);
+        checkFluidStack(helper, sourceTank.getFluid(), Fluids.WATER, 4000);
+        checkItemStack(helper, targetStack, Items.BUCKET, 2);
+        helper.succeed();
+    }
+
+    /**
+     * Ensures that tryEmptyContainerAndStow doesn't duplicate fluids in the target when the container is stackable.
+     * Regression test for <a href="https://github.com/MinecraftForge/MinecraftForge/issues/6796">issue #6796</a>.
+     */
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Legacy tryEmptyContainerAndStow_stackable")
+    private static void test_tryEmptyContainerAndStow_stackable(ExtendedGameTestHelper helper) {
```

Comment:
This should be "Ensures that tryEmptyContainerAndStow doesn't duplicate fluids in the target when the container is stackable."